zerver/views/upload.py

```diff
@@ -25,6 +27,20 @@
 from zerver.models import UserProfile, validate_attachment_request
 
 
+def patch_disposition_header(response: HttpResponse, url: str, is_attachment: bool) -> None:
+    disposition = "attachment" if is_attachment else "inline"
+
+    filename = os.path.basename(urlparse(url).path)
+
+    try:
+        filename.encode("ascii")
+        file_expr = 'filename="{}"'.format(filename.replace("\\", "\\\\").replace('"', r"\""))
+    except UnicodeEncodeError:
+        file_expr = "filename*=utf-8''{}".format(quote(filename))
```

Comment:
This encoding algorithm is fairly involved; would it make sense to add a comment?

Reply:
Added a few comments.

Reply:
Done.